### ISSUE TITLE
elm init fails for no apparent reason

### ISSUE THREAD

**Issue by andrejcremoznik** [elm/compiler#2024]
**Quick Summary:**

Running `elm init` fails.

- **Elm:** 0.19.1 (linux 64 binary)
- **Operating System:** Linux 5.3.12 (Arch)

## Additional Details

```
-- PROBLEM LOADING PACKAGE LIST ------------------------------------------------

I need the list of published packages before I can start initializing projects,
so I tried to fetch:

    https://package.elm-lang.org/all-packages

But my HTTP library is giving me the following error message:

    ConnectionFailure Network.Socket.getAddrInfo (called with preferred socket type/protocol: AddrInfo {addrFlags = [AI_ADDRCONFIG], addrFamily = AF_UNSPEC, addrSocketType = Stream, addrProtocol = 0, addrAddress = <assumed to be undefined>, addrCanonName = <assumed to be undefined>}, host name: Just "package.elm-lang.org", service name: Just "443"): does not exist (Try again)
```

1. I can load the URL through browser just fine
2. I can `curl` the URL through the same shell session just fine
3. I have FTTH, it is not slow. But the server seems to be on the another side of the planet because ping is 106ms. There is no packet loss.
4. Pretty sure firewall and pi.hole can't be the reason if I can access the URL otherwise just fine.

I must say this is a pretty crappy first experience. If only there was some debug flag to produce more connection details. My guess is the embedded http lib doesn't play well with dns/resolving/tls or something. I'd compile from source, but couldn't find any build instructions.

**Comment by rlefevre** [comment]
`elm` binary uses `/etc/resolv.conf` for DNS resolution.
Systems using `systemd` may have this file linking to `stub-resolv.conf` wich for some network configurations may contain only some private DNS servers, which does not allow DNS requests to succeed.

Check and remember where your `/etc/resolv.conf` points to, then try:
```
$ sudo rm -f /etc/resolv.conf
$ sudo ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
```
Revert back if it does not work.

**Comment by andrejcremoznik** [comment]
Ok, it worked, but the resolution was a bit different;

I use `systemd-networkd` and `systemd-resolved` together. `/etc/resolv.conf` was the default, empty text file. I symlinked that to `/run/systemd/resolve/stub-resolv.conf` and restarted the `systemd-resolved` service and it worked.

Looking at [Arch wiki](https://wiki.archlinux.org/index.php/Systemd-resolved#DNS), this is recommended, but I overlooked it. For some reason nothing else until now raised any resolv errors.
